src/cv/aggregator.cpp

```diff
@@ -0,0 +1,95 @@
+#include "cv/aggregator.hpp"
+#include "utilities/locks.hpp"
+#include "utilities/constants.hpp"
+#include "utilities/logging.hpp"
+#include "utilities/lockptr.hpp"
+
+CVAggregator::CVAggregator(Pipeline p): pipeline{p} {
+    this->num_worker_threads = 0;
+    this->results = std::make_shared<CVResults>();
+    for (int i = 0; i < this->results->matches.size(); i++) {
+        this->results->matches[i] = -1;
+    }
+}
+
+CVAggregator::~CVAggregator() {
+
+}
+
+LockPtr<CVResults> CVAggregator::getResults() {
+    return LockPtr<CVResults>(this->results, &this->mut);
+}
+
+void CVAggregator::runPipeline(const ImageData& image) {
+    Lock lock(this->mut);
+
+    if (this->num_worker_threads > MAX_CV_PIPELINES) {
+        LOG_F(WARNING, "Tried to spawn more than %ld CVAggregator workers at one time.", MAX_CV_PIPELINES);
+        this->overflow_queue.push(image);
+        LOG_F(WARNING, "CVAggregator overflow queue at size %ld", this->overflow_queue.size());
+        return;
+    }
+
+    static int thread_num = 0;
```

Comment:
Reviewing C++ is so difficult. I would have never caught it if you got this `static` wrong, or any other locking or memory issue.